### PR TITLE
Multipoint constraints

### PR DESCRIPTION
**Pull request by adityaghantasala**
Addresses error reported in #970 

- Removing erroneous recursive call of serializer in MpcData class. 
- custom load and save functions or MpcData object. 
- Printing information in << operator overload

### CODE REVIEW

**Review by roigcarlo — APPROVED** [review]
Looks fine to me 👍

File: applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp [hunk 1]
```diff
@@ -244,10 +234,10 @@ class MpcData
     {
         std::cout << std::endl;
         std::cout << "===============================================================" << std::endl;
-        std::cout << "Number of Slave DOFs :: "<< mDofConstraints.size()<<std::endl;
-        for(auto i : mDofConstraints)
+        std::cout << "Number of Slave DOFs :: " << mDofConstraints.size() << std::endl;
+        for (auto i : mDofConstraints)
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp]: const

File: applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp [hunk 2]
```diff
@@ -285,29 +273,60 @@ class MpcData
 
     virtual void save(Serializer &rSerializer) const
     {
-        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, MpcData);
+        rSerializer.save("MpcDataName", mName);
+        rSerializer.save("NumConstraints", mDofConstraints.size());
+        for (auto& slaveMasterrelation : mDofConstraints){
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp]: const?

File: applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp [hunk 3]
```diff
@@ -285,29 +273,60 @@ class MpcData
...
-        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, MpcData);
+        rSerializer.save("MpcDataName", mName);
+        rSerializer.save("NumConstraints", mDofConstraints.size());
+        for (auto& slaveMasterrelation : mDofConstraints){
+
+            rSerializer.save("slaveID", (slaveMasterrelation.first).first); // saving the vector of the slave id 
+            rSerializer.save("slaveKey", (slaveMasterrelation.first).second); // saving the vector of the slave key
+
+            rSerializer.save("numMasters", (slaveMasterrelation.second).size()); // Writint number of masters for this slave
+            for ( auto masterIdKeyConstant: (slaveMasterrelation.second) ){
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp]: const ref?

File: applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp (hunk 1 above)
Comment (adityaghantasala) [reply applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp]: Changed in latest commit.

**Pull request merged**